File: java/rna-transcription/src/main/java/RnaTranscription.java
import java.util.HashMap;
import java.util.Map;

public class RnaTranscription {

    private static final Map<Character, Character> rnaMap = createRnaMap();
    private static Map<Character, Character> createRnaMap() {
        Map<Character, Character> map = new HashMap<Character, Character>();
        map.put('C', 'G');
        map.put('G', 'C');
        map.put('T', 'A');
        map.put('A', 'U');
        return map;
    }

    public String transcribe(String dnaStrand) {
        if (dnaStrand == null || dnaStrand.trim().isEmpty()) return "";
        char[] dnaChars = dnaStrand.toCharArray();
        StringBuilder rnaBuilder = new StringBuilder();
        for (Character dnaChar : dnaChars) {
            if (rnaMap.containsKey(dnaChar))
                rnaBuilder.append(rnaMap.get(dnaChar));
        }
        return rnaBuilder.toString();
    }
}

File: csharp/rna-transcription/RnaTranscription.cs
using System.Collections.Generic;
using System.Linq;

public static class RnaTranscription
{
  private static readonly Dictionary<char, char> RNA_MAP = new Dictionary<char, char>()
  {
    { 'C', 'G' },
    { 'G', 'C' },
    { 'T', 'A' },
    { 'A', 'U' }
  };

  public static string ToRna(string dnaStrand)
  {
    return new string(dnaStrand
      .Where(x => RNA_MAP.ContainsKey(x))
      .Select(x => RNA_MAP[x])
      .ToArray());
  }
}